cloud/scope/machine.go

```diff
@@ -766,6 +771,99 @@ func (m *MachineScope) DeleteVPCLoadBalancerPoolMember() error {
 	return nil
 }
 
+// deleteVPCLoadBalancerPoolMembers provides support to delete Load Balancer Pools Members for a Machine that are tracked in the Machine's Status, which is part of the extended VPC Machine support.
+// This new support allows a Machine to have members in multiple Load Balancers, as defined by the Machine Spec, rather than defaulting (legacy) to the single Cluster Load Balancer.
+func (m *MachineScope) deleteVPCLoadBalancerPoolMembers() error {
+	cleanupIncomplete := false
+	for _, member := range m.IBMVPCMachine.Status.LoadBalancerPoolMembers {
+		// Retrieve the Instance details immediately (without them the member cannot be safely deleted).
+		instanceOptions := &vpcv1.GetInstanceOptions{
+			ID: ptr.To(m.IBMVPCMachine.Status.InstanceID),
+		}
+		instanceDetails, _, err := m.IBMVPCClient.GetInstance(instanceOptions)
+		if err != nil {
+			return fmt.Errorf("error retrieving instance for load balancer pool member deletion for machine %s: %w", m.IBMVPCMachine.Name, err)
+		}
+		// Verify the instance has a primary network interface IP address needed to delete the correct Load Balancer Pool Member.
+		if instanceDetails.PrimaryNetworkInterface == nil || instanceDetails.PrimaryNetworkInterface.PrimaryIP == nil || instanceDetails.PrimaryNetworkInterface.PrimaryIP.Address == nil {
+			return fmt.Errorf("error instance is missing the primary network interface IP address for load balancer pool member deletion for machine: %s", m.IBMVPCMachine.Name)
+		}
+		m.Logger.V(5).Info("collected instance details for load balancer pool member deletion", "machienName", m.IBMVPCMachine.Name, "instanceID", *instanceDetails.ID, "instanceIP", *instanceDetails.PrimaryNetworkInterface.PrimaryIP.Address)
```

Comment:
Can thins also moved out of for loop? We are fetching machine details and it remains same?

Reply:
Yep, will move.

Reply:
done